third_party/nvfuser/csrc/scheduler/registry.cpp

```diff
@@ -1839,6 +1842,202 @@ class PersistentKernelScheduler : public SchedulerEntry {
+          max_multi_reduction_factor,
+          ", is smaller than minimum multi reduction factor, ",
+          min_multi_reduction_factor);
+      return false;
+    }
+
+    const int64_t max_used_sms = is_cross_grid
+        ? ceilDiv(
+              ceilDiv(properties.total_iteration_numel, vectorization_factor),
+              cross_grid_params->launch_params.bdimx()) *
+            cross_grid_params->launch_params.gdimy()
+        : ceilDiv(
+              properties.total_iteration_numel * persistent_buffer_size,
+              sm_register_file_size);
+
+    // Bandwidth suffers if the number of used SMs is small. This is
+    // particularly impactful in the case of cross grid, so at least
+    // half of the SMs are required to be used. In the case of cross
+    // block, keep using the existing heuristics for now.
+    if (is_cross_grid &&
+        max_used_sms <
+            scheduler_utils::safeDiv(device_multiprocessor_count, 2)) {
+      scheduler_debug_utils::canScheduleRejectReason(
+          ScheduleHeuristic::Persistent,
+          "cross grid - not enough used SMs: ",
+          max_used_sms);
+      return false;
+    }
+
+    const int64_t device_max_threads_per_multiprocessor =
+        (int64_t)device_prop->maxThreadsPerMultiProcessor;
+    const int64_t min_fraction_of_sms =
+        scheduler_utils::safeDiv(device_multiprocessor_count, 8);
+    if (properties.total_reduction_numel >=
+            device_max_threads_per_multiprocessor * 4 && // Large reduction dim
+        max_used_sms < min_fraction_of_sms) {
+      scheduler_debug_utils::canScheduleRejectReason(
+          ScheduleHeuristic::Persistent, "not enough used SMs");
+      return false;
+    }
+
+    // The runtime kernel for grouped normal grid reductions is not
+    // well tuned, and it turned out to be quite difficult to get
+    // consistently better performances than non-persistent
+    // schedules. Disabled for now.
+    // TODO: Enable non-welford persistent reductions
+    if (is_cross_grid &&
+        std::any_of(
+            reduction_tvs.begin(),
+            reduction_tvs.end(),
+            [](TensorView* reduction_tv) {
+              return !reduction_tv->definition()->isA<WelfordOp>();
+            })) {
+      scheduler_debug_utils::canScheduleRejectReason(
+          ScheduleHeuristic::Persistent, "non-Welford not enabled yet");
+      return false;
+    }
+
+    // Had a hard time tuning on Titan RTX and V100 when the iteration
+    // space is not evenly divided by threads and thread blocks. It
+    // doesn't seem to be noticeably bad on A100, though. For now,
+    // disable the schedule if not evenly divisible just on Titan RTX
+    // only.
+    // TODO: Revisit
+    if (is_cross_grid &&
+        (properties.total_iteration_numel %
+             (vectorization_factor * cross_grid_params->launch_params.bdimx() *
+              cross_grid_params->launch_params.gdimx()) !=
+         0) &&
+        device_prop->major == 7) {
```

Comment:
Isn't this disabling this on more than just titan rtx? Don't you need a minor version here?

Reply:
Missed updating the last sentence of the comment above, but as mentioned in the first line, I had this issue with both Titan RTX and V100, so I just disabled anything with major version number of 7.

Reply:
Clarified the comment.